src/Catty/ViewController/Upload/UploadViewController.swift

```diff
@@ -56,46 +39,46 @@ class UploadViewController: UIViewController {
 
     private var zipFileData: Data?
     private var loadingView: LoadingView?
-    private var _session: URLSession?
-    private var session: URLSession? {
-        if _session == nil {
-            // Initialize Session Configuration
-            let sessionConfiguration = URLSessionConfiguration.default
-
-            // Configure Session Configuration
-            sessionConfiguration.httpAdditionalHeaders = ["Accept": "application/json"]
-
-            // Initialize Session
-            _session = URLSession(configuration: sessionConfiguration)
-        }
-        return _session
-    }
 
     // MARK: - View Lifecycle
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        view.backgroundColor = UIColor.background
-        initProjectNameViewElements()
-        initSizeViewElements()
-        initDescriptionViewElements()
-        initObservers()
-        hideKeyboardWhenTapInViewController()
-
-        self.uploadBarButton = UIBarButtonItem(title: kLocalizedUploadProject,
-                                               style: .plain,
-                                               target: self,
-                                               action: #selector(UploadViewController.checkProjectAction))
-        navigationItem.rightBarButtonItem = self.uploadBarButton
-
-        projectNameTextField.becomeFirstResponder()
+        if self.nibName != nil {
+            view.backgroundColor = UIColor.background
+            initProjectNameViewElements()
+            initSizeViewElements()
+            initDescriptionViewElements()
+            initObservers()
+            hideKeyboardWhenTapInViewController()
+
+            self.uploadBarButton = UIBarButtonItem(title: kLocalizedUploadProject,
+                                                   style: .plain,
+                                                   target: self,
+                                                   action: #selector(UploadViewController.checkProjectAction))
+            navigationItem.rightBarButtonItem = self.uploadBarButton
+
+            projectNameTextField.becomeFirstResponder()
+        }
     }
 
     // MARK: - Initialization
 
     required init?(coder aDecoder: NSCoder) {
         super.init(coder: aDecoder)
         self.project = Project.init(loadingInfo: Util.lastUsedProjectLoadingInfo())!
+        self.initUploader()
+    }
+
+    init(uploader: StoreProjectUploaderProtocol) {
+        activeRequest = true
+        self.project = Project()
+        super.init(nibName: nil, bundle: nil)
+        initUploader(uploader: uploader)
+    }
+
+    func initUploader(uploader: StoreProjectUploaderProtocol = StoreProjectUploader(fileManager: CBFileManager())) {
```

Comment:
Please remove this method and set the default value directly in the required `init` (since it is only of use there).